src/basicfuns.jl

```diff
@@ -441,3 +441,74 @@ $(SIGNATURES)
 Compute the complementary double exponential, `1 - exp(-exp(x))`.
 """
 cexpexp(x) = -_expm1(-exp(x))
+
+#=
+this uses the identity:
+
+log(logistic(x)) = -log(1 + exp(-x))
+=#
+"""
+$(SIGNATURES)
+
+Return `log(logistic(x))`, computed more carefully and with fewer calls
+than the naive composition of functions.
+
+Its inverse is the [`logitexp`](@ref) function.
+"""
+loglogistic(x::Real) = _loglogistic(x)
+_loglogistic(x::Real) = -log1pexp(-x)
+_loglogistic(x::Union{Integer, Rational}) = _loglogistic(float(x))
```

Comment:
I'm still not sure whether unsigned integers are worth any consideration given that you will run into problems with them in many (most?) calculations and definitions in JuliaStats, JuliaMath, etc. typically do not handle them in a special way - but if we care then it would be easier to just define
```suggestion
loglogistic(x::Real) = -log1pexp(-float(x))
```
This would cover also cases such as e.g. unitful integer numbers that are not covered by `Union{Integer, Rational}` and for floating points such as `x::Float64` or `x::Float32` it would be equivalent to `-log1pexp(-x)`.

Reply:
I opted for the `float` solution, as this is simplest and, furthermore, replicates the handling of integer and rational arguments in Julia `Base`.

> I'm still not sure whether unsigned integers are worth any consideration

SpecialFunctions.jl (another package I contributed to) converts most arguments via `float`, hence, it handles both the unsigned integer cases as well as the signed integer (i.e. `typemin(T<:Signed)`) edge case.

Use of `float` to convert integers and rationals to a floating point type for use in special functions is common throughout `Base`. Unless integers and rationals are converted, then a function interface defined as `x::Real` cannot be claimed. On the other hand, if the intention is to guarantee reasonable results only for floating point numbers, then this should clearly be stated in the package description. However, I am of the opinion that this package should follow the behavior of `Base` and `SpecialFunctions`, whereby integers and rationals are converted to floating point numbers internally.

Unless conversion is enforced, this leads to some obviously incorrect results, e.g., `logistic(typemin(Int)) == 1.0`, but, clearly, should be `0.0`.